toolkit/tools/pkg/imagecustomizerlib/customizeos.go

```diff
@@ -30,6 +30,20 @@ func doOsCustomizations(ctx context.Context, rc *ResolvedConfig, imageConnection
 		return err
 	}
 
+	// If UKI is enabled, extract kernel and initramfs from existing UKIs for re-customization.
+	if rc.Config.OS.Uki != nil {
```

Comment:
What happens when the base image has UKIs but the user doesn't specify UKI in the config file?

Reply:
1. During validation phase, the `connectToExistingImage()` is trying to read kernel cmdline args from `grub.cfg` but failed, then falls back to extract from UKI, which is fine.
2. `ensureGrubCfgForUki()` won't be called, no grub.cfg available.
3. At package phase, if new Kernel installed / updated, won't impact the UKI / bootloader, in addition, the kernel remove will cause error.
4. Bootloader handling phase, it might regenerate grub.cfg anyway, but the grub.cfg and UKI will be in inconsistent state I think.

Maybe I need to detect base image has UKI but config doesn't specify it then fail with error. Thoughts?

Reply:
Take a look at the `TestCustomizeImageVerityUsrUki` test. The `verity-reinit-usr-nop.yaml` config currently relies on the config not specifying the UKI field as a way of ensuring the UKI is not updated/refreshed, which would invalidate any signature on the UKI. We need to ensure this scenario continues to work. But we can change the API if necessary.

I think it might make sense to require the user to explicitly tell us how to handle the base image's UKI.

Something like:

```yaml
uki:
  reinitialize: ["" | "passthrough" | "refresh"]
```

Where:

- `""`: Error if the base image has UKIs
- `"passthrough"`: Ensure the existing UKIs are used. Don't allow new UKIs.
- `"refresh"`: Delete the existing UKIs and regenerate them from scratch.